### PR TITLE
 Fix date double escaping #342

### PR DESCRIPTION
**Pull request by freddyboucher**

### CODE REVIEW

File: restygwt/src/main/java/org/fusesource/restygwt/rebind/RestServiceClassCreator.java [hunk 1]
```diff
@@ -936,7 +936,11 @@ protected String toStringExpression(JType type, String expr) throws UnableToComp
 
         if (DATE_TYPE == type) {
             String encoded = locator.encodeExpression(DATE_TYPE, expr, Style.DEFAULT);
-            return "java.util.Objects.toString(" + encoded + ", null)";
+            return String.format(
+                    "%s instanceof com.google.gwt.json.client.JSONString ?" +
+                            " ((com.google.gwt.json.client.JSONString) %s).stringValue() :" +
+                            " java.util.Objects.toString(%s, null)",
```
Comment (mkristian) [inline restygwt/src/main/java/org/fusesource/restygwt/rebind/RestServiceClassCreator.java]: I never saw this Objects class so I look up the javadocs and it turns out that `java.util.Objects.toString(%s)` would already do the right thing ;)

File: restygwt/src/test/java/org/fusesource/restygwt/client/basic/DirectRestServiceTestGwt.java [hunk 2]
```diff
@@ -107,6 +108,102 @@ public void onFailure(Method method, Throwable e) {
...
+
+    public void testDateFormatCustomPatternNull() {
+        delayTestFinish(10000);
+        DirectExampleService directExampleService = GWT.create(DirectExampleService.class);
+
+        Resource resource = new Resource(GWT.getModuleBaseURL() + "api");
+        ((RestServiceProxy) directExampleService).setResource(resource);
+
+        // custom pattern contains double quote and single quote to check JSON escaping
+        Defaults.setDateFormat("\"''yyyy-MM-dd'T'HH:mm:ss.SSSZZZ");
```
Comment (mkristian) [inline restygwt/src/test/java/org/fusesource/restygwt/client/basic/DirectRestServiceTestGwt.java]: 👍

File: restygwt/src/main/java/org/fusesource/restygwt/rebind/RestServiceClassCreator.java (hunk 1 above)
Comment (freddyboucher) [reply restygwt/src/main/java/org/fusesource/restygwt/rebind/RestServiceClassCreator.java]: Nop it's not the same:
`Objects.toString(null)` returns the String`"null"`
`Objects.toString(null, null)` returns `null`

Comment (mkristian) [reply restygwt/src/main/java/org/fusesource/restygwt/rebind/RestServiceClassCreator.java]: oh. cool

**Pull request merged**